fbinternal/cloud/go/services/testcontroller/statemachines/enodebd.go

```
@@ -335,13 +444,23 @@ func trafficTest(trafficTestNumber int, stateNumber int, gatewayClient GatewayCl
 
 	resp, err := gatewayClient.GenerateTraffic(*config.NetworkID, trafficGWID, config.Ssid, config.SsidPw)
 	if resp == nil || err != nil {
+		if successState == subscriberActiveState {
+			pretext = fmt.Sprintf(trafficInactiveSubPretextFmt, *config.SubscriberID, *config.EnodebSN, *config.AgwConfig.TargetGatewayID, "SUCCEEDED")
+			postToSlack(machine.client, *config.AgwConfig.SLACKWebhook, false, pretext, fallback, "", "")
+			return successState, 1 * time.Minute, err
+		}
 		if stateNumber >= maxTrafficStateCount {
 			pretext = fmt.Sprintf(trafficPretextFmt, trafficTestNumber, *config.EnodebSN, *config.AgwConfig.TargetGatewayID, "FAILED")
 			postToSlack(machine.client, *config.AgwConfig.SLACKWebhook, false, pretext, fallback, "", "")
 			return checkForUpgradeState, 1 * time.Minute, errors.Errorf("Traffic test number %d failed on gwID %s after %d tries", trafficTestNumber, trafficGWID, maxTrafficStateCount)
 		}
 		return fmt.Sprintf(trafficTestStateFmt, trafficTestNumber, stateNumber+1), 1 * time.Minute, err
 	}
+	if successState == subscriberActiveState {
+		pretext = fmt.Sprintf(trafficInactiveSubPretextFmt, *config.SubscriberID, *config.EnodebSN, *config.AgwConfig.TargetGatewayID, "FAILED")
+		postToSlack(machine.client, *config.AgwConfig.SLACKWebhook, true, pretext, fallback, "", "")
+		return checkForUpgradeState, 1 * time.Minute, err
```

`err` here is always `nil`. We should instead return an error here. This make more intuitive sense by returning an error anytime the test doesn't succeed (i.e. it's weird to return an error despite the test achieving the expected behavior as we have currently in line 450).